beacon-chain/state/state-native/getters_validator.go

```diff
@@ -180,6 +180,22 @@ func (b *BeaconState) PubkeyAtIndex(idx primitives.ValidatorIndex) [fieldparams.
 	return bytesutil.ToBytes48(v.PublicKey)
 }
 
+// PublicKeys builds a list of all validator public keys, with each key's index aligned to its validator index.
+func (b *BeaconState) PublicKeys() [][fieldparams.BLSPubkeyLength]byte {
+	b.lock.RLock()
+	defer b.lock.RLock()
+
+	res := make([][fieldparams.BLSPubkeyLength]byte, len(b.validators))
```

Comment:
with multi value slice feature being merged in, we would also need to add support for it over here.

Reply:
Please take another look - updated with a snippet from @rkapka